### ISSUE TITLE
CoCalc Instructor: add collect and hide from student option

### ISSUE THREAD

**Issue by profbrady** [sagemathinc/cocalc#3663]
When using CoCalc for giving tests I would like an option to collect assignments and hide the associated students' assignment folders from their view until the assignment is returned. This would keep students from showing their test to students in other sections so the test questions don't get out into the wild. I know that students who are driven to share their work (i.e. cheat) will do so by taking screen shots or some other method, but this might stop the "casual" cheater.

**Comment by williamstein** [comment]
This could be challenging to implement, since it is impossible to build on top of our current security model.   What we could do instead is have a special complete project *just for this particular assignment*, and remove the student from the project after the homework is collected -- that fits the security model, and would solve your problem.

**Comment by profbrady** [comment]
That could work. Currently I am collecting individually from every student in particular section then running a script that deletes the assignment folder for all students. I then assign for the students in the next session, collect, and delete the folder for everybody again. Probably not the best method, but seems to be working okay today.

**Comment by williamstein** [comment]
> that deletes the assignment folder for all students.

If they really want to, they can just click the "Backups" button, and navigate to the snapshot of the folder before you deleted it.   Alternatively, they can recreate the file, then use timetravel to go back to when it wasn't empty.

**Comment by profbrady** [comment]
Yes they could, but since I haven’t explicitly shown then how to do that, most would not think of it. Like I said, those willing to find a way to copy/cheat probably will. I am trying to make it “out of sight, out of mind.”
